packages/typespec-ts/src/index.ts

```diff
@@ -166,9 +180,7 @@ export async function $onEmit(context: EmitContext) {
         buildModels(modularCodeModel, project, modularSourcesRoot, subfolder);
         buildModelsOptions(subClient, project, modularSourcesRoot, subfolder);
         const hasClientUnexpectedHelper =
-          needUnexpectedHelper.get(
-            subClient.rlcClientName.replace("Context", "Client")
-          ) ?? false;
+          needUnexpectedHelper.get(subClient.rlcClientName) ?? false;
```

Comment:
Small fix: the set and get client name is inconsistancy in `needUnexpectedHelper`.